tests/test_models.py

```diff
@@ -782,6 +782,23 @@ def test_logo_path(self, file_exists, delete_called):
             if delete_called:
                 storage_mock.delete.assert_called_once_with('enterprise/branding/1/1_logo.png')
 
+    def test_logo_path_after_save(self):
+        """
+        Test that `EnterpriseCustomerBrandingConfiguration`.logo is saved at the correct path with correct id after the
+        model instance is saved.
+        """
+        branding_config = EnterpriseCustomerBrandingConfiguration(
+            enterprise_customer=factories.EnterpriseCustomerFactory(),
+            logo="logo.png"
+        )
+        storage_mock = mock.MagicMock(spec=Storage, name="StorageMock")
+        with mock.patch("django.core.files.storage.default_storage._wrapped", storage_mock):
+            branding_config.save()
+            config_id = branding_config.id
+            path = logo_path(branding_config, branding_config.logo.name)
+            expected_path = "enterprise/branding/"+str(config_id)+"/"+str(config_id)+"_logo.png"
```

Comment:
I think we should assert the saved path of the image with the expected path from the logo_path.
```
saved_path = branding_config.logo.path
expected_path = logo_path(branding_config, branding_config.logo.name)
self.assertEqual(saved_path, expected_path)
```

Reply:
I can't check the exact path as I'm mocking Storage. All I need to test is that the path has the correct instance id in it instead of None.

Reply:
I didn't get why we have to mock storage for this test?

Reply:
Let's catch up on Slack